source/lac/trilinos_sparse_matrix.cc

```diff
@@ -2449,16 +2449,33 @@ namespace TrilinosWrappers
 {
   namespace internal
   {
+    namespace
+    {
+#ifndef DEAL_II_WITH_MPI
```

Comment:
I don't think you can link a serial Trilinos built with a parallel built of PETSc. We only allow a serial built of Trilinos if deal.II is non-MPI. (Serial Trilinos is somewhat of a side feature that we do not test extensively. I have one copy on my computer but I hardly ever use it. We probably should have never introduced it because it requires explicitly distinguishing between the `Communicator` objects as you observed.) I'm not sure about the PETSc configuration options we offer, but I don't think we can have PETSc without MPI.